sdk/analytics/src/main/java/com/klaviyo/analytics/model/Profile.kt

```diff
@@ -14,21 +14,27 @@ class Profile(properties: Map<ProfileKey, Serializable>?) :
     var externalId: String?
         get() = (this[ProfileKey.EXTERNAL_ID])?.toString()
         set(value) {
-            this[ProfileKey.EXTERNAL_ID] = value
```

Comment:
i don't think you need to change `Profile` at all. This is just a data class, if for some reason you need to set email back to null you can. This doesn't affect state.